pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -299,3 +310,41 @@ func removeSSHRule(ctx context.Context, cl k8sClient.Client, infraID string) err
 
 	return nil
 }
+
+// PostDestroy deletes the ignition bucket after capi stopped running, so it won't try to reconcile the bucket.
+func (p *Provider) PostDestroy(ctx context.Context, in clusterapi.PostDestroyerInput) error {
+	region := in.Metadata.AWS.Region
+	session, err := awsconfig.GetSessionWithOptions(
+		awsconfig.WithRegion(region),
+		awsconfig.WithServiceEndpoints(region, in.Metadata.AWS.ServiceEndpoints),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create aws session: %w", err)
+	}
+
+	bucketName := fmt.Sprintf("openshift-bootstrap-data-%s", in.Metadata.InfraID)
+	if err := removeS3Bucket(ctx, session, bucketName); err != nil {
+		if p.bestEffortDeleteIgnition {
```

Comment:
I should probably check specifically for authorization issues here but still return errors in the other cases. Will fix once I have feedback on the overall approach.

Reply:
> Will fix once I have feedback on the overall approach.

overall approach looks good. everything I see that I want changed is not from this pr :D 

> I should probably check specifically for authorization issues here but still return errors in the other cases. 

Hm I'm not sure it's just auth errors that we want to skip. I believe the intention may be to skip errors in the case of s3 versioning. I'm not sure how those errors present.

Reply:
We could replicate the destroy code and try to delete the object versions first.